### ISSUE TITLE
Electron overriding GNOME icon

### ISSUE THREAD

**Issue by cortopy** [electron/electron#5241]
I have set the icon property for `new BrowserWindow()` . When I run the app via comman `electron` the icon is shown OK. However, this doesn't work when launched with a GNOME app.desktop file

Unlike Windows, there doesn't seem to be a specific method for setting the icon in a GNOME environment. So I have created an app.desktop file. This works fine during launch as the icon is shown as it should. However, once the app is ready, the icon is overriden by the same generic system icon I would get if no icon was set at all.

In short, when app is launched directly, icon property works fine. However, when app is launched with app.desktop file, the icon set in that file is overriden by a generic one.

* Electron version: 0.37.6
* Operating system: Fedora 23

**Comment by zcbenz** [comment]
On Linux how the icon is showed is purely depended on the desktop environment, the `icon` property passed by Electron is just a hint for the DE. But if you are creating a correct `.desktop` file, the icon should show normally, you take Atom's file for example:
https://github.com/atom/atom/blob/f48069c5f848b5ae624b110a1df7800e919eb142/resources/linux/atom.desktop.in

**Comment by cortopy** [comment]
That's why I opened this issue. The .desktop file has been validated with `desktop-file-validate`. The icon is shown during bootstrap, but once the app is ready, Electron overrides the icon with a generic one.

**Comment by mkaz** [comment]
@cortopy we've seen the same issue with the WordPress.com app and never quite figured out why

**Comment by zcbenz** [comment]
@mkaz You can probably try using absolute path for `Icon` in `.desktop` file.

**Comment by mkaz** [comment]
@zcbenz Thanks, I think I've got it figured out. We were missing it in a config on our Linux builds, also found that using the same name for app and icon helps.

For the curious, here's the PR that should fix it our app https://github.com/Automattic/wp-desktop/pull/183
